torch/testing/_internal/common_methods_invocations.py

```diff
@@ -8341,6 +8348,21 @@ def gradcheck_wrapper_triangular_input(op, *args, upper=False, idx=0, **kwargs):
     return op(*args[:idx], triangular_arg, *args[idx + 1:], upper, **kwargs)
 
 
+def gradcheck_wrapper_triangular_input_real_positive_diagonal(op, *args, upper=False, idx=0, **kwargs):
+    """Gradcheck wrapper for functions that take lower/upper triangular matrices
+    with real and positive diagonals, for example, cholesky-like operations.
+    """
+    arg = args[idx]
+    arg_diag = arg.diagonal(0, -2, -1)
+    arg_diag_embed = torch.diag_embed(arg_diag)
+    # new_arg = arg - diag(arg) + I
+    new_arg = arg - arg_diag_embed + torch.diag_embed(torch.ones_like(arg_diag))
```

Comment:
Looks like the diag is always ones here? Is that okay or do we want to do random positive values?